tests/serializers/test_serialize_as_any.py

```diff
@@ -155,3 +156,38 @@ class Other:
         'x': 1,
         'y': 'hopefully not a secret',
     }
+
+
+def test_serialize_with_recursive_models() -> None:
+
+    class Node:
+        next: Optional['Node'] = None
+        value: int = 42
+
+    schema = core_schema.definitions_schema(
```

Comment:
@sydney-runkle I tested the python code and all good but how do I make an equivalent `schema` for the test here ? 

```python
from __future__ import annotations

import traceback

import pydantic


class Model1(pydantic.BaseModel):
    recursive: Model1 | None


class Model2(pydantic.BaseModel):
    recursive: pydantic.SerializeAsAny[Model2] | None

print(Model1(recursive=None).model_dump())
print(Model1(recursive=None).model_dump_json())
print(Model2(recursive=None).model_dump())
print(Model2(recursive=None).model_dump_json())

# All fail:
try:
    print(Model1(recursive=None).model_dump(serialize_as_any=True))
except:
    traceback.print_exc()

try:
    print(Model1(recursive=None).model_dump_json(serialize_as_any=True))
except:
    traceback.print_exc()

try:
    print(Model2(recursive=None).model_dump(serialize_as_any=True))
except:
    traceback.print_exc()

try:
    print(Model2(recursive=None).model_dump_json(serialize_as_any=True))
except:
    traceback.print_exc()

```

Reply:
Good question - I normally just do 
```py
from pydantic._internal._core_utils import pretty_print_core_schema
from pydantic import BaseModel

class SomeModel(BaseModel): ...

pretty_print_core_schema(SomeModel.__pydantic_core_schema__)
```

And then go from that representation in order to build up an equivalent core schema